### PR TITLE
Rename kubernetes salt module and fix deps

### PR DESCRIPTION
**Pull request by sayf-eddine-scality**
https://github.com/saltstack/salt/pull/51936
Fixes: #923

### CODE REVIEW

File: buildchain/buildchain/salt_tree.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
     Path('salt/_modules/containerd.py'),
     Path('salt/_modules/cri.py'),
     Path('salt/_modules/docker_registry.py'),
-    Path('salt/_modules/kubernetes.py'),
+    Path('salt/_modules/kubernetesmod.py'),
```
Comment (slaperche-scality) [inline buildchain/buildchain/salt_tree.py:212]: You should explain in the commit message WHY we must rename the file.

Comment (slaperche-scality) [inline buildchain/buildchain/salt_tree.py:212]: Also, not very fond of `kubernetesmod`, maybe `k8s_utils` or sth like that would be better

File: salt/_modules/metalk8s.py [hunk 2]
```diff
@@ -4,16 +4,19 @@
 '''
 import logging
 
+from salt.exceptions import CommandExecutionError
```
Comment (slaperche-scality) [inline salt/_modules/metalk8s.py:7]: Those changes seem totally unrelated, should be done in a dedicated commit (or dedicated PR).

Or am I wrong and it is related to the renaming?

**Review by slaperche-scality — CHANGES_REQUESTED** [review]

**Review by gdemonet — DISMISSED** [review]
LGTM

File: buildchain/buildchain/salt_tree.py (hunk 1 above)
Comment (gdemonet) [reply buildchain/buildchain/salt_tree.py:212]: A link to the Saltstack PR/issue could be enough IMHO

Comment (slaperche-scality) [reply buildchain/buildchain/salt_tree.py:212]: Ok, I heard the reason for behind this name so let's keep it.

Comment (slaperche-scality) [reply buildchain/buildchain/salt_tree.py:212]: Yes an no, I really would like to have some info that are readable offline.

**Review by cmonfort — DISMISSED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Review by slaperche-scality — DISMISSED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e approve

**Review by TeddyAndrieux — DISMISSED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e reset

**Review by TeddyAndrieux — APPROVED** [review]
